### PR TITLE
Malfunctioning AI Blackout + RCD Disable Ability Tweaks

### PR DESCRIPTION
**Pull request by gurfan**
The blackout and RCD disable abilities for malfunctioning AIs are really underwhelming and overpriced.

Currently they do the following:

Blackout costs 15 points and gives you a three use spell that when cast, has a ~30% to overload the lighting on each of the station's APCs. The chances increases slightly each time it fails.
RCD disable costs a whopping 50 points and makes currently-existing RCD unusuable. That's it.

This PR merges the two abilities and tweaks them a bit:

The RCD Disable ability is removed, it is now part of Blackout.
Blackout only has one use per purchase, rather than three.
When used, Blackout does the following:

- Overloads the lighting of *every* APC on the station.
- Disables any currently-existing RCDs (new ones that are built are unaffected)
- Causes any currently-existing radios to have receiving messages scrambled. (New radios that are built are unaffected)

Affected radios will have any messages that they receive turned into mostly-unintelligible gibberish. 
Ex.  OH MY GOD AI ROGUE -> O*%^ $GO% A% !RG%E@
Binary chat is unaffected, of course.

The goal of this change was to give the AI a good disruption tool that goes beyond what can normally be achieved with standard AI abilities.

[tweak]
[balance]

:cl:
 * rscadd: The Blackout Malf Module will now disable RCDs and scramble radios on the station in addition to breaking lights.
 * rscdel: Removed the RCD Disable Malf Module (Now part of Blackout)

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
I remember that RCDs are specifically disabled because people would use them to break through the walls pretty easily but I'm not sure it can affect reinforced walls anymore

**Comment by gurfan** [comment]
RCDs cant deconstruct reinforced walls anymore, only doors and regular walls.

**Comment by BomberBro** [comment]
Wait, it had a 30% chance of blackout for an APC... when there's already a button for ALL AIs to overload the lights on any APC?

**Comment by gurfan** [comment]
> Wait, it had a 30% chance of blackout for an APC... when there's already a button for ALL AIs to overload the lights on any APC?

Correct.

**Comment by Blithering** [comment]
Maybe it should scramble hand teles and uncalibrate telesci too, to keep in the spirit of the intended "stop high tech tools from allowing walking right into the core" idea

**Comment by gurfan** [comment]
> Maybe it should scramble hand teles and uncalibrate telesci too, to keep in the spirit of the intended "stop high tech tools from allowing walking right into the core" idea

Telesci can already be easily uncalibrated by the AI, and since people can't teleport into the core without a bluespace crystals its more of a tool for the AI than the crew. I don't see how hand teles can come in to play without setting them to 'random', which is already risky and luck-based.

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -215,16 +191,22 @@ rcd light flash thingy on matter drain
...
 	for(var/obj/machinery/power/apc/apc in power_machines)
-		if(prob(30*apc.overload))
-			apc.overload_lighting()
-		else
-			apc.overload++
+		apc.overload_lighting()
+	to_chat(world, radio_list.len)
+	for(var/obj/item/device/radio/R in radio_list)
+		R.scramble_message = 1
+	for(var/obj/item/device/rcd/matter/engineering/rcd in rcd_list)
```
Comment (jknpj) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Honestly just make RCDs check and break themselves instead, the way it is done currently is pretty much just shitcode(not your fault).

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 2]
```diff
@@ -215,16 +191,22 @@ rcd light flash thingy on matter drain
...
 
 /spell/aoe_turf/blackout/cast(var/list/targets, mob/user)
 	for(var/obj/machinery/power/apc/apc in power_machines)
-		if(prob(30*apc.overload))
-			apc.overload_lighting()
-		else
-			apc.overload++
+		apc.overload_lighting()
+	to_chat(world, radio_list.len)
+	for(var/obj/item/device/radio/R in radio_list)
```
Comment (jknpj) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Why isn't tcomms handling this, jesus christ malf code...

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 3]
```diff
@@ -215,16 +191,22 @@ rcd light flash thingy on matter drain
...
+		R.scramble_message = 1
+	for(var/obj/item/device/rcd/matter/engineering/rcd in rcd_list)
+		rcd.disabled = 1
+	for(var/obj/item/mecha_parts/mecha_equipment/tool/red/red in red_tool_list)
```
Comment (jknpj) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Ignore that, i'm retarded

Comment (jknpj) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: The RED still uses an RCD insternally so making them break themselves can potentially solve this, too.

**Comment by SonixApache** [comment]
ask jknpjr if you need to make changes and if so, make the changes
else alert the collabs

**Comment by Kurfursten** [comment]
did this get the changes it needed?

**Pull request merged**